compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -226,6 +226,37 @@ void quant_bias_per_channel(CircleConst *node, float input_scale, std::vector<fl
   }
 }
 
+void int16_quant_bias_per_tensor(CircleConst *node, float input_scale,
+                                 std::vector<float> &weight_scale,
+                                 std::vector<float> &scaling_factor)
+{
+  uint32_t size = node->size<loco::DataType::FLOAT32>();
+  std::vector<int64_t> quantized_values(size);
+
+  auto abs_compare = [](float x, float y) { return std::abs(x) < std::abs(y); };
+  const float weight_scale_max =
+      *std::max_element(weight_scale.begin(), weight_scale.end(), abs_compare);
+  assert(weight_scale_max >= 0);
+  scaling_factor[0] = input_scale * weight_scale_max;
+
+  for (uint32_t i = 0; i < size; ++i)
+  {
+    double scaling_factor_inv = (scaling_factor[0] == 0) ? 0 : 1.0 / scaling_factor[0];
+    quantized_values[i] =
+        static_cast<int64_t>(std::round(node->at<loco::DataType::FLOAT32>(i) * scaling_factor_inv));
+  }
+
+  node->dtype(loco::DataType::S64);      // change the type of tensor
+  node->size<loco::DataType::S64>(size); // resize tensor
+  const int64_t kMinScale = std::numeric_limits<int64_t>::lowest();
+  const int64_t kMaxScale = std::numeric_limits<int64_t>::max();
+  for (uint32_t i = 0; i < size; ++i)
+  {
+    node->at<loco::DataType::S64>(i) =
```

Comment:
This clamping seems unnecessary as the `quantized_values` is vector<int64_t>.
Not sure about the lowest bound though, should it be -2^32 or -(2^32-1)?
In any case this loop could be fused into the loop above :)

Reply:
Clamping seems unncessary. I've updated the code.

It is not simple to fuse the loops because the type/shape of node is changed between the loops.